src/payment/data/reducers.js

```diff
@@ -17,58 +17,55 @@ const basketInitialState = {
-      ...state,
-      paymentMethod: action.payload.method,
-    };
-    case submitPayment.REQUEST: return {
-      ...state,
-      submitting: true,
-      isBasketProcessing: true,
-    };
-    case submitPayment.SUCCESS: return {
-      ...state,
-      redirect: true,
-    };
-    case submitPayment.FULFILL: return {
-      ...state,
-      submitting: false,
-      isBasketProcessing: false,
-      paymentMethod: undefined,
-    };
-
-    case fetchBasket.TRIGGER: return { ...state, loading: true };
-    case fetchBasket.SUCCESS: return { ...state, data: action.payload };
-    case fetchBasket.FAILURE: return { ...state, error: action.payload };
-    case fetchBasket.FULFILL: return { ...state, loading: false, loaded: true };
-
-    case addCoupon.REQUEST: return { ...state, isBasketProcessing: true };
-    case addCoupon.SUCCESS: return { ...state, couponData: action.payload };
-    case addCoupon.FAILURE: return { ...state, couponError: action.payload };
-    case addCoupon.FULFILL: return { ...state, isBasketProcessing: false };
-
-    case removeCoupon.REQUEST: return { ...state, isBasketProcessing: true };
-    case removeCoupon.SUCCESS: return { ...state, couponData: action.payload };
-    case removeCoupon.FAILURE: return { ...state, couponError: action.payload };
-    case removeCoupon.FULFILL: return { ...state, isBasketProcessing: false };
-
-    case updateQuantity.REQUEST: return { ...state, isBasketProcessing: true };
-    case updateQuantity.SUCCESS: return { ...state, quantityData: action.payload };
-    case updateQuantity.FAILURE: return { ...state, quantityError: action.payload };
-    case updateQuantity.FULFILL: return { ...state, isBasketProcessing: false };
-
-    default:
-      return state;
+  if (action !== null) {
+    switch (action.type) {
+      case BASKET_DATA_RECEIVED: return { ...state, ...action.payload };
+
+      case submitPayment.TRIGGER: return {
+        ...state,
+        paymentMethod: action.payload.method,
+      };
+      case submitPayment.REQUEST: return {
+        ...state,
+        submitting: true,
+        isBasketProcessing: true,
+      };
+      case submitPayment.SUCCESS: return {
+        ...state,
+        redirect: true,
+      };
+      case submitPayment.FULFILL: return {
+        ...state,
+        submitting: false,
+        isBasketProcessing: false,
+        paymentMethod: undefined,
+      };
+
+      case fetchBasket.TRIGGER: return { ...state, loading: true };
+      case fetchBasket.FULFILL: return { ...state, loading: false, loaded: true };
+
+      case addCoupon.REQUEST:
+      case removeCoupon.REQUEST:
+      case updateQuantity.REQUEST:
+        return { ...state, isBasketProcessing: true };
+
+      case addCoupon.FULFILL:
+      case removeCoupon.FULFILL:
+      case updateQuantity.FULFILL:
+        return { ...state, isBasketProcessing: false };
+
+      default:
+    }
   }
+  return state;
 };
 
+/* istanbul ignore next */
```

Comment:
Mocking universal-cookie isn't possible, far as I can tell. And node/jsdom doesn't have cookies.  So this code is basically untestable.